app/src/component/mediaPlayer/view.js

```diff
@@ -261,6 +261,12 @@ class MediaPlayer extends React.PureComponent {
     return null;
   }
 
+  getEncodedDownloadPath = (fileInfo) => {
+    const { file_name: fileName } = fileInfo;
+    const encodedFileName = encodeURIComponent(fileName).replace(/!/g, '%21');
```

Comment:
The entire set of unescaped characters is `-_.!~*'()`, does this need to reflect that?

Reply:
Not necessarily. The rest of the characters can be left as is in the filename, but if we have to escape any of them in the future, we could just add an additional `replace` call here.